compiler/rustc_trait_selection/src/traits/project.rs

```diff
@@ -611,8 +615,12 @@ impl<'a, 'b, 'tcx> TypeFolder<TyCtxt<'tcx>> for AssocTypeNormalizer<'a, 'b, 'tcx
                 );
                 normalized_ty
             }
-
-            _ => ty.super_fold_with(self),
+            ty::Weak => self
+                .selcx
+                .tcx()
+                .type_of(data.def_id)
+                .subst(self.selcx.tcx(), data.substs)
+                .fold_with(self),
```

Comment:
Shouldn't we be instantiating bounds here and adding them to `self.obligations`?

Reply:
Yes, but everything I tried to test didn't have an effect, because we already do that due to the opaque type inside of the type alias already adding those obligations